src/components/Button/Button.js

```diff
@@ -1,25 +1,29 @@
 import React from 'react';
 import './Button.scss';
 import PropTypes from 'prop-types';
+import classNames from 'classnames';
 
 function Button(props) {
   const { variant, children, size = 'small' } = props;
+
   return (
     <button
-      role="button"
       type="button"
       value="button"
-      className={`button ${variant} ${size}`}
+      className={`button ${classNames(
+        { [variant]: variant },
+        { [size]: size }
+      )}`}
```

Comment:
The whole point of using `classnames` is to not have to deal with string interpolation. A good example of its use case can be found [here](https://github.com/JedWatson/classnames#usage-with-reactjs), in the project README.

In this case, `classnames` could be used like this:

```js
classNames('button', variant, size)
```

If `variant` or `size` are `undefined`, they will not be joined to the output class name string. That's the key takeaway here, as the library puts it:

> If the value associated with a given key is falsy, that key won't be included in the output.

I tend to assign the generated class name to a variable and then use that variable as the component prop. This keeps the code cleaner, in my opinion, and makes JSX more readable. For example:

```jsx
export const Button = (props) => {
  const className = classNames('button', props.variant, props.size);

  return <button className={className}>Submit</button>;
}
```

---


The object argument approach is clumsy here because the desired class names match the `variant` and `size` values. If the component used boolean values to determine its classes, then we'd have something much more useful. For example:

```js
classNames('button', { small: props.isSmall, medium: props.isMedium, large: props.isLarge });
```